app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/LocalFileTransferActivity.java

```diff
@@ -0,0 +1,437 @@
+    return true;
+  }
+
+  @Override
+  public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+    if (item.getItemId() == R.id.menu_item_search_devices) {
+
+      /* Permissions essential for this module */
+      if (!checkCoarseLocationAccessPermission()) {
+        return true;
+      }
+
+      if (!checkExternalStorageWritePermission()) {
+        return true;
+      }
+
+      /* Initiate discovery */
+      if (!wifiDirectManager.isWifiP2pEnabled()) {
+        requestEnableWifiP2pServices();
+        return true;
+      }
+
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && !isLocationServicesEnabled()) {
+        requestEnableLocationServices();
+        return true;
+      }
+
+      showPeerDiscoveryProgressBar();
+      wifiDirectManager.discoverPeerDevices();
+
+      return true;
+    } else {
+      return super.onOptionsItemSelected(item);
+    }
+  }
+
+  @Override
+  public void onUserDeviceDetailsAvailable(@Nullable WifiP2pDevice userDevice) { // Update UI with user device's details
+    if (userDevice != null) {
+      deviceName.setText(userDevice.deviceName);
+      Log.d(TAG, getDeviceStatus(userDevice.status));
+    }
+  }
+
+  @Override
+  public void clearListOfAvailablePeers() {
+    availablePeerDevices.clear();
+    ((WifiPeerListAdapter) listViewPeerDevices.getAdapter()).notifyDataSetChanged();
+  }
+
+  private void displayFileTransferProgress(@NonNull ArrayList<FileItem> filesToSend) {
+    fileListAdapter = new FileListAdapter(filesToSend);
+    filesRecyclerView.setAdapter(fileListAdapter);
+    filesRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+  }
+
+  @Override
+  public void onFilesForTransferAvailable(@NonNull ArrayList<FileItem> filesForTransfer) {
+    this.filesForTransfer = filesForTransfer;
+    displayFileTransferProgress(filesForTransfer);
+  }
+
+  private void showPeerDiscoveryProgressBar() { // Setup UI for searching peers
+    searchingPeersProgressBar.setVisibility(View.VISIBLE);
+    listViewPeerDevices.setVisibility(View.INVISIBLE);
+    textViewPeerDevices.setVisibility(View.INVISIBLE);
+  }
+
+  @Override
+  public void onSenderStartedConnection() {
+    this.hasSenderStartedConnection = true;
+  }
+
+  @Override
+  public void onFileStatusChanged(int itemIndex) {
+    fileListAdapter.notifyItemChanged(itemIndex);
+  }
+
+  @Override
+  public void updateListOfAvailablePeers(@NonNull WifiP2pDeviceList peers) {
+    availablePeerDevices.clear();
+    availablePeerDevices.addAll(peers.getDeviceList());
+
+    searchingPeersProgressBar.setVisibility(View.GONE);
+    listViewPeerDevices.setVisibility(View.VISIBLE);
+    ((WifiPeerListAdapter) listViewPeerDevices.getAdapter()).notifyDataSetChanged();
+
+    if (availablePeerDevices.size() == 0) {
+      Log.d(LocalFileTransferActivity.TAG, "No devices found");
+    }
+  }
+
+  @Override
+  public void onFileTransferComplete(boolean wereAllFilesTransferred) {
```

Comment:
seems like this boolean is pointless

Reply:
So there's a `onFileTransferAsyncTaskComplete(boolean wereAllFilesTransferred)` method in WDM (which gets a value for that boolean from the async-tasks), which then calls this method.
As of now I'm handling display of error/success toasts in WDM itself, so should I change the signature here then?

Reply:
If you don't need it then don't have it, if another impl needs it later on add it back in.
Less code is best code.

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/f071ad60b7716d7c548a5d2a13f510c45482d01d#diff-e1d59ae6e15dcb502e125fb6e243d15dR234)